pkg/prom/cleaner_test.go

```diff
@@ -0,0 +1,170 @@
+package prom
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/grafana/agent/pkg/prom/instance"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Basic logger to write to stderr to help debug test failures
+func getLogger() log.Logger {
+	return log.NewLogfmtLogger(os.Stderr)
+}
```

Comment:
nit: I'd rather not define an extra function here. It adds more to think about when reading the code ("what kind of logger does getLogger return?"). To avoid the repetition, you can just define the logger at the top of the tests:

```go
logger := log.NewLogfmtLogger(os.Stderr)
cleaner := NewWALCleaner(logger, &mockManager{}, walRoot)
```

(I can be convinced otherwise though)

Reply:
Sounds good